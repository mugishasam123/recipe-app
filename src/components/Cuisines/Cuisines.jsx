import Card from '../Card/Card'

const Cuisines = ({ filtered }) => {
  
  return (
    <div>
      {filtered.length === 0 ? (
        <div className="flex justify-center">
          <h2 className="md:text-xl">No Cuisines found</h2>
        </div>
      ) : (
        filtered.map(({ id, title, image, sourceUrl }) => (
          <Card
            key={id}
            id={id}
            title={title}
            image={image}
            sourceUrl={sourceUrl}
          />
        ))
      )}
    </div>
  )
}

export default Cuisines
