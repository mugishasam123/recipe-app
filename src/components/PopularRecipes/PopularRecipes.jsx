import { useEffect, useState } from 'react'
import Card from '../Card/Card'

import { Splide, SplideSlide } from '@splidejs/react-splide'
import '@splidejs/splide/dist/css/themes/splide-default.min.css'

const PopularRecipes = () => {
  const [random, setRandom] = useState([])

  const fetchRandom = async () => {
    const resp = await fetch(
      `https://api.spoonacular.com/recipes/random?apiKey=ed210e26302d4f00910371959212ad1f&number=10`,
    )
    const data = await resp.json()
    setRandom(data.recipes)
    localStorage.setItem('random', JSON.stringify(data.recipes))
    return data.recipes
  }

  useEffect(() => {
    const randomRecipe = localStorage.getItem('random')
    if (randomRecipe) {
      setRandom(JSON.parse(randomRecipe))
    } else {
      fetchRandom()
    }
  }, [])

  return (
    <div className="mx-7">
      <h2 className="m-5 mt-1 font-bold text-lg">Popular Picks</h2>
      <Splide
        options={{
          perPage: 3,
          arrows: true,
          pagination: false,
          drag: 'free',
          gap: '4rem',
          breakpoints: {
            1024: {
              perPage: 3,
            },
            767: {
              perPage: 2,
            },
            640: {
              perPage: 1,
            },
          },
        }}
      >
        {random.map(({ title, id, image, sourceUrl }) => (
          <SplideSlide key={id}>
            <Card id={id} title={title} image={image} sourceUrl={sourceUrl} />
          </SplideSlide>
        ))}
      </Splide>
    </div>
  )
}

export default PopularRecipes
