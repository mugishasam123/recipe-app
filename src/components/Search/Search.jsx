import { useState } from 'react'
import { FaSearch } from 'react-icons/fa'
import { useNavigate } from 'react-router-dom'

const Search = () => {
  const [searchTerm, setSearchTerm] = useState('')
  const navigate = useNavigate()

  const submitHandler = (e) => {
    e.preventDefault()
    navigate(`/searched/${searchTerm}`)
  }
  return (
    <form className="my-10" onSubmit={submitHandler}>
      <div className="relative min-w-min md:w-[50%] lg:w-[50%] m-auto w-[80%]">
        <FaSearch className="absolute top-[30%] mx-2" />
        <div className="flex">
          <input
            className="bg-gray-200 text-lg py-2 w-full outline-none rounded-md shadow-sm placeholder:text-slate-400 focus:ring-gray-600 focus:ring-1 pl-9 pr-3"
            type="text"
            placeholder="Search recipe.."
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
          />
          <button type="submit" className="bg-gray-500 p-4 rounded mx-2">
            Search
          </button>
        </div>
      </div>
    </form>
  )
}

export default Search
