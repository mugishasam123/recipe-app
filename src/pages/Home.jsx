import PopularRecipes from '../components/PopularRecipes/PopularRecipes'
import Search from '../components/Search/Search'
import SearchResults from '../components/SearchResults/SearchResults'
import { motion } from 'framer-motion'
import { Link } from 'react-router-dom'
import { GiKnifeFork } from 'react-icons/gi'

const Home = () => {
  return (
    <motion.div>
      <div>
        <div className="flex justify-start items-center p-5">
          <GiKnifeFork className="text-2xl" />
          <Link to={'/'} className="text-xl font-medium">
            Recipe app
          </Link>
        </div>
      </div>
      <PopularRecipes />
      <Search />
      <SearchResults />
    </motion.div>
  )
}

export default Home
