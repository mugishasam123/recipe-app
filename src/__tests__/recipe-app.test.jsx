import { describe, expect, test, vi, Vitest } from 'vitest'
import { Route, Routes } from 'react-router-dom'
import { fireEvent, screen, waitFor } from '@testing-library/react'
import { render } from '@testing-library/react'
import Home from '../../src/pages/Home'
import PopularRecipes from '../components/PopularRecipes/PopularRecipes'
import Cuisines from '../components/Cuisines/Cuisines'
import Search from '../components/Search/Search'

const mockData = [
  {
    id: 1,
    title: 'Recipe 1',
    image: 'recipe1.jpg',
    imageType: 'jpg',
    sourceUrl:
      'http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers',
  },
  {
    id: 2,
    title: 'Recipe 2',
    image: 'recipe2.jpg',
    imageType: 'jpg',
    sourceUrl:
      'http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers',
  },
  {
    id: 3,
    title: 'Recipe 3',
    image: 'recipe3.jpg',
    imageType: 'jpg',
    sourceUrl:
      'http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers',
  },
]

describe('Component rendering', () => {
  test('testing if Cuisines is rendered', () => {
    expect(render(<Cuisines filtered={mockData} />)).toMatchSnapshot()
  })
})

describe('UI', () => {
  test('testing if Cuisines show recipe title', () => {
    render(<Cuisines filtered={mockData} />)
    expect(screen.getByText('Recipe 1')).toBeTruthy
  })
})
