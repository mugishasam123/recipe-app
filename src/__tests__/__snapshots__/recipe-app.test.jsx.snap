// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component rendering > testing if Cuisines is rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
          target="_blank"
        >
          <div
            class="min-h-[15rem] overflow-hidden relative rounded-lg"
          >
            <p
              class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
            >
              Recipe 1
            </p>
            <img
              alt="Recipe 1"
              class="absolute left-0 w-full h-[100%] object-cover"
              src="recipe1.jpg"
            />
            <div
              class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
            />
          </div>
        </a>
        <a
          href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
          target="_blank"
        >
          <div
            class="min-h-[15rem] overflow-hidden relative rounded-lg"
          >
            <p
              class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
            >
              Recipe 2
            </p>
            <img
              alt="Recipe 2"
              class="absolute left-0 w-full h-[100%] object-cover"
              src="recipe2.jpg"
            />
            <div
              class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
            />
          </div>
        </a>
        <a
          href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
          target="_blank"
        >
          <div
            class="min-h-[15rem] overflow-hidden relative rounded-lg"
          >
            <p
              class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
            >
              Recipe 3
            </p>
            <img
              alt="Recipe 3"
              class="absolute left-0 w-full h-[100%] object-cover"
              src="recipe3.jpg"
            />
            <div
              class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
        target="_blank"
      >
        <div
          class="min-h-[15rem] overflow-hidden relative rounded-lg"
        >
          <p
            class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
          >
            Recipe 1
          </p>
          <img
            alt="Recipe 1"
            class="absolute left-0 w-full h-[100%] object-cover"
            src="recipe1.jpg"
          />
          <div
            class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
          />
        </div>
      </a>
      <a
        href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
        target="_blank"
      >
        <div
          class="min-h-[15rem] overflow-hidden relative rounded-lg"
        >
          <p
            class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
          >
            Recipe 2
          </p>
          <img
            alt="Recipe 2"
            class="absolute left-0 w-full h-[100%] object-cover"
            src="recipe2.jpg"
          />
          <div
            class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
          />
        </div>
      </a>
      <a
        href="http://www.foodista.com/recipe/JKGPWDDP/steak-with-lemon-and-capers"
        target="_blank"
      >
        <div
          class="min-h-[15rem] overflow-hidden relative rounded-lg"
        >
          <p
            class="absolute inset-x-0 z-10 text-gray-200 bottom-0 h-16 w-full text-center font-bold"
          >
            Recipe 3
          </p>
          <img
            alt="Recipe 3"
            class="absolute left-0 w-full h-[100%] object-cover"
            src="recipe3.jpg"
          />
          <div
            class="bg-gradient-to-t from-gray-900 to-transparent absolute w-full h-[100%] z-3 opacity-90"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
